hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/AbstractBlockNodeConnection.java

```diff
@@ -212,6 +213,20 @@ void onTerminalStateTransition() {
         return configuration;
     }
 
+    /**
+     * Given a throwable, determine if the throwable or one of its causes is a {@link GrpcException}.
+     *
+     * @param t the throwable to search
+     * @return the {@link GrpcException} associated with this throwable, or null if one is not found
+     */
+    protected GrpcException findGrpcException(final Throwable t) {
+        if (t instanceof final GrpcException ge) {
+            return ge;
+        }
+        final Throwable cause = t.getCause();
+        return cause == null ? null : findGrpcException(cause);
```

Comment:
Iterating will be slightly faster here than recursion, but if performance isn't a concern, sure 👍